filebeat/input/kafka/input.go

```diff
@@ -104,9 +105,14 @@ func NewInput(
 func (input *kafkaInput) runConsumerGroup(
 	context context.Context, consumerGroup sarama.ConsumerGroup,
 ) {
+
 	handler := &groupHandler{
 		version: input.config.Version,
 		outlet:  input.outlet,
+		// yieldEventsFromField will be assigned the configuration option yield_events_from_field,
+		// if the fileset using this input expects to receive multiple messages bundled under a specific field then this value is assigned
+		// ex. in this case are the azure fielsets where the events are found under the json object "records"
+		yieldEventsFromField: input.config.YieldEventsFromField,
```

Comment:
I like this new approach! I think that in order to speed things up you could send a PR with the changes required in the input first and then one for the module. WDYT?

Reply:
opened https://github.com/elastic/beats/pull/13965